internal/cmd/marketplace/apply.go

```diff
@@ -0,0 +1,319 @@
+)
+
+func listFilesInDirRecursive(rootPath string) ([]string, error) {
+	var files []string
+	err := filepath.Walk(rootPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return files, nil
+}
+
+func buildPathsListFromDir(dirPath string) ([]string, error) {
+	filesPaths, err := listFilesInDirRecursive(dirPath)
+	if err != nil {
+		return nil, err
+	}
+	filePaths := []string{}
+	for _, path := range filesPaths {
+		switch filepath.Ext(path) {
+		case encoding.YamlExtension:
+			fallthrough
+		case encoding.YmlExtension:
+			fallthrough
+		case encoding.JSONExtension:
+			filePaths = append(filePaths, path)
+		default:
+			return nil, fmt.Errorf("%w: %s", errInvalidExtension, path)
+		}
+	}
+	return filePaths, nil
+}
+
+func buildApplyRequest(pathList []string) (*ApplyRequest, error) {
+	resources := []Item{}
+	resNameToFilePath := map[string]string{}
+	for _, filePath := range pathList {
+		content, err := os.ReadFile(filePath)
+		if err != nil {
+			return nil, err
+		}
+		var fileEncoding string
+		switch filepath.Ext(filePath) {
+		case encoding.YamlExtension:
+			fallthrough
+		case encoding.YmlExtension:
+			fileEncoding = YAML
+		case encoding.JSONExtension:
+			fileEncoding = JSON
+		default:
+			return nil, fmt.Errorf("%w: %s", errInvalidExtension, filePath)
+		}
+		marketplaceItem := &Item{}
+		err = encoding.UnmarshalData(content, fileEncoding, marketplaceItem)
+		if err != nil {
+			return nil, fmt.Errorf("errors in file %s: %w", filePath, err)
+		}
+		resName, err := retrieveAndValidateResName(*marketplaceItem, resNameToFilePath, filePath)
+		if err != nil {
+			return nil, err
+		}
+		resources = append(resources, *marketplaceItem)
+		resNameToFilePath[resName] = filePath
+	}
+	if len(resources) == 0 {
+		return nil, errNoValidFilesProvided
+	}
+	return &ApplyRequest{
+		Resources: resources,
+	}, nil
+}
+
+func retrieveAndValidateResName(res Item, resNameToFilePath map[string]string, filePath string) (string, error) {
+	resName, ok := res["name"]
+	if !ok {
+		return "", fmt.Errorf("%w: %s", errResWithoutName, filePath)
+	}
+	resNameStr, ok := resName.(string)
+	if !ok {
+		return "", fmt.Errorf("%w: %s", errResNameNotAString, filePath)
+	}
+	if _, ok := resNameToFilePath[resNameStr]; ok {
+		return "", fmt.Errorf("%w: %s", errDuplicatedResName, resName)
+	}
+
+	return resNameStr, nil
+}
+
+func applyMarketplaceResource(ctx context.Context, client *client.APIClient, companyID string, request *ApplyRequest) (*ApplyResponse, error) {
+	if companyID == "" {
+		return nil, errors.New("companyID must be defined")
+	}
+
+	bodyBytes, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Post().
+		APIPath(fmt.Sprintf(applyEndpoint, companyID)).
+		Body(bodyBytes).
+		Do(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
You are missing the check if the response contains an error from the server. The first error only contains the network ones.

```suggestion
	if err := resp.Error(); err != nil {
		return nil, err
	}
```